---
# Documentation: https://wowchemy.com/docs/managing-content/

title: Microanatomy of the human atherosclerotic plaque by single-cell transcriptomics
subtitle: ''
summary: ''
authors:
- Marie A C Depuydt
- Koen H M Prange
- Lotte Slenders
- Tiit Örd
- Danny Elbersen
- Arjan Boltjes
- Saskia C A de Jager
- Folkert W Asselbergs
- Gert J de Borst
- Einari Aavik
- Tapio Lönnberg
- Esther Lutgens
- Christopher K Glass
- Hester M den Ruijter
- Minna U Kaikkonen
- Ilze Bot
- Bram Slütter
- Sander W van der Laan
- Seppo Yla-Herttuala
- Michal Mokry
- Johan Kuiper
- Menno P J de Winther
- Gerard Pasterkamp
tags:
- atherosclerosis; cardiovascular disease; genome-wide association study; single-cell
  analysis
categories: []
date: '2020-11-01'
lastmod: 2022-01-11T15:28:38+01:00
featured: false
draft: false

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder.
# Focal points: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight.
image:
  caption: ''
  focal_point: ''
  preview_only: false

# Projects (optional).
#   Associate this post with one or more of your projects.
#   Simply enter your project's folder or file name without extension.
#   E.g. `projects = ["internal-project"]` references `content/project/deep-learning/index.md`.
#   Otherwise, set `projects = []`.
projects: []
publishDate: '2022-01-11T14:28:38.648472Z'
publication_types:
- '2'
abstract: 'RATIONALE: Atherosclerotic lesions are known for their cellular heterogeneity,
  yet the molecular complexity within the cells of human plaques has not been fully
  assessed. OBJECTIVE: Using single-cell transcriptomics and chromatin accessibility,
  we gained a better understanding of the pathophysiology underlying human atherosclerosis.
  METHODS AND RESULTS: We performed single-cell RNA and single-cell ATAC sequencing
  on human carotid atherosclerotic plaques to define the cells at play and determine
  their transcriptomic and epigenomic characteristics. We identified 14 distinct cell
  populations including endothelial cells, smooth muscle cells, mast cells, B cells,
  myeloid cells, and T cells and identified multiple cellular activation states and
  suggested cellular interconversions. Within the endothelial cell population, we
  defined subsets with angiogenic capacity plus clear signs of endothelial to mesenchymal
  transition. CD4+ and CD8+ T cells showed activation-based subclasses, each with
  a gradual decline from a cytotoxic to a more quiescent phenotype. Myeloid cells
  included 2 populations of proinflammatory macrophages showing IL (interleukin) 1B
  or TNF (tumor necrosis factor) expression as well as a foam cell-like population
  expressing TREM2 (triggering receptor expressed on myeloid cells 2) and displaying
  a fibrosis-promoting phenotype. ATACseq data identified specific transcription factors
  associated with the myeloid subpopulation and T cell cytokine profiles underlying
  mutual activation between both cell types. Finally, cardiovascular disease susceptibility
  genes identified using public genome-wide association studies data were particularly
  enriched in lesional macrophages, endothelial, and smooth muscle cells. CONCLUSIONS:
  This study provides a transcriptome-based cellular landscape of human atherosclerotic
  plaques and highlights cellular plasticity and intercellular communication at the
  site of disease. This detailed definition of cell communities at play in atherosclerosis
  will facilitate cell-based mapping of novel interventional targets with direct functional
  relevance for the treatment of human disease.'
publication: '*Circ. Res.*'
---
